invenio_accounts/views/rest.py

```diff
@@ -0,0 +1,533 @@
+
+    decorators = [user_already_authenticated]
+
+    post_args = {
+        'token': fields.String(required=True),
+        'password': fields.String(
+            required=True, validate=[validate.Length(min=6, max=128)]),
+    }
+
+    def get_user(self, token=None, **kwargs):
+        """Retrieve a user by the provided arguments."""
+        # Verify the token
+        expired, invalid, user = reset_password_token_status(token)
+        if invalid:
+            _abort(get_message('INVALID_RESET_PASSWORD_TOKEN')[0])
+        if expired:
+            _abort(get_message(
+                'PASSWORD_RESET_EXPIRED',
+                email=user.email,
+                within=current_security.reset_password_within)[0])
+        return user
+
+    def success_response(self, user):
+        """Return a successful reset password response."""
+        return jsonify({'message': get_message('PASSWORD_RESET')[0]})
+
+    @use_kwargs(post_args)
+    def post(self, **kwargs):
+        """Reset user password."""
+        user = self.get_user(**kwargs)
+        after_this_request(_commit)
+        update_password(user, kwargs['password'])
+        login_user(user)
+        return self.success_response(user)
+
+
+class ChangePasswordView(MethodView):
+    """View to change the user password."""
+
+    decorators = [login_required]
+
+    post_args = {
+        'password': fields.String(
+            required=True, validate=[validate.Length(min=6, max=128)]),
+        'new_password': fields.String(
+            required=True, validate=[validate.Length(min=6, max=128)]),
+    }
+
+    def verify_password(self, password=None, new_password=None, **kwargs):
+        """Verify password is not invalid."""
+        if not verify_and_update_password(password, current_user):
+            _abort(get_message('INVALID_PASSWORD')[0], 'password')
+        if password == new_password:
+            _abort(get_message('PASSWORD_IS_THE_SAME')[0], 'password')
+
+    def change_password(self, new_password=None, **kwargs):
+        """Perform any change password actions."""
+        after_this_request(_commit)
+        change_user_password(user=current_user._get_current_object(),
+                             password=new_password)
+
+    def success_response(self):
+        """Return a successful change password response."""
+        return jsonify({'message': get_message('PASSWORD_CHANGE')[0]})
+
+    @use_kwargs(post_args)
+    def post(self, **kwargs):
+        """Change user password."""
+        self.verify_password(**kwargs)
+        self.change_password(**kwargs)
+        return self.success_response()
+
+
+class SendConfirmationEmailView(MethodView):
+    """View function which sends confirmation instructions."""
+
+    decorators = [login_required]
+
+    confirmation_link_func = default_confirmation_link_func
+
+    post_args = {
+        'email': fields.Email(required=True, validate=[user_exists]),
+    }
+
+    def get_user(self, email=None, **kwargs):
```

Comment:
This function is repeated at last twice, so it could be placed in a Mixin class instead to avoid duplication.